### ISSUE TITLE
Hard monotonic model validation loss

### ISSUE THREAD

**Issue by kylebgorman** [CUNY-CL/yoyodyne#317]
Hard monotonic models compute loss as a side effect of the training forward pass. During the validation forward pass though they greedily decode and this does not directly produce loss. We currently take the negative means of the likelihoods returned by this decoding as validation loss, but this is not directly comparable to training loss (they have rather different magnitudes).

The validation step should be refined so that it produces a loss compatible with training, even if it means a bit more work. (This is already a very fast model at least in the sense it converges very rapidly in my experience.) Note that with this fix it'll probably be possible to no longer return the likelihood tensor from greedy decoding (instead we'll probably return loss here?). 

PS: It would be great to have beam search for this (#17).
